project/src/main/webapp/event-details.js

```diff
@@ -91,14 +91,23 @@ function getLinkForOpportunity(opportunityId) {
 }
 
 /**
- * Gets event details from database with eventId and uses to fill event page
+ * Gets event details from database with eventId and fills out event page with details
+ * If registering for event, register user then show event details
  */
 async function getEventDetails() {
+    //make sign up link go to correct
   const queryString = window.location.search;
   const urlParams = new URLSearchParams(queryString);
   const eventId = urlParams.get('eventId');
-  const response = await fetch('/create-event?' +
-    new URLSearchParams({'eventId': eventId}));
+
+  //Register for event, if register param not specified, default to not registering
+  const registerBool = urlParams.get('register') ? urlParams.get('register'): 'false';
```

Comment:
This part confuses me a bit, because it seems the same as if you one-lined it and said `if (urlParams.get('register'))` or `if (urlParams.get('register')) === 'true'`